RecoParticleFlow/PFTracking/src/PFCheckHitPattern.cc

```diff
@@ -72,24 +67,34 @@ void PFCheckHitPattern::init(edm::ESHandle<TrackerGeometry> tkerGeomHandle_) {
 #endif
 }
 
-PFCheckHitPattern::DetInfo PFCheckHitPattern::interpretDetId(DetId detId) {
+PFCheckHitPattern::DetInfo PFCheckHitPattern::interpretDetId(DetId detId, const TrackerTopology* tkerTopo) {
   // Convert detId to a pair<uint32, uint32> consisting of the numbers used by HitPattern 
   // to identify subdetector and layer number respectively.
-  if (detId.subdetId() == StripSubdetector::TIB) {
-    return DetInfo( detId.subdetId() , TIBDetId(detId).layer() );
-  } else if (detId.subdetId() == StripSubdetector::TOB) {
-    return DetInfo( detId.subdetId() , TOBDetId(detId).layer() );
-  } else if (detId.subdetId() == StripSubdetector::TID) {
-    return DetInfo( detId.subdetId() , TIDDetId(detId).wheel() );
-  } else if (detId.subdetId() == StripSubdetector::TEC) {
-    return DetInfo( detId.subdetId() , TECDetId(detId).wheel() );
-  } else if (detId.subdetId() == PixelSubdetector::PixelBarrel) {
-    return DetInfo( detId.subdetId() , PXBDetId(detId).layer() );
-  } else if (detId.subdetId() == PixelSubdetector::PixelEndcap) {
-    return DetInfo( detId.subdetId() , PXFDetId(detId).disk() );
-  } else {
-    throw cms::Exception("RecoParticleFlow", "Found DetId that is not in Tracker");
-  }   
+  const auto subdetId = detId.subdetId();
+  uint32_t second{};
+  switch (subdetId) {
+    case StripSubdetector::TIB:
+      second = tkerTopo->tibLayer(detId);
+      break;
+    case StripSubdetector::TOB:
+      second = tkerTopo->tobLayer(detId);
+      break;
+    case StripSubdetector::TID:
+      second = tkerTopo->tidWheel(detId);
+      break;
+    case StripSubdetector::TEC:
+      second = tkerTopo->tecWheel(detId);
+      break;
+    case PixelSubdetector::PixelBarrel:
+      second = tkerTopo->pxbLayer(detId);
+      break;
+    case PixelSubdetector::PixelEndcap:
+      second = tkerTopo->pxfDisk(detId);
+      break;
+    default:
+      throw cms::Exception("RecoParticleFlow", "Found DetId that is not in Tracker");
+  }
+  return DetInfo(subdetId, second);
```

Comment:
I believe this could be simplified to
```
return DetInfo(detId.subdetId(), tkerTopo->layer(detId));
```
Or maybe avoid this function altogether (AFAICT there is only a single call to this function)? e.g. in line 35
```
auto detId = dets[i]->geographicalId();
auto detInfo = DetInfo(detId.subdetId(), tkerTopo->layer(detId));
```


https://github.com/cms-sw/cmssw/blob/master/DataFormats/TrackerCommon/src/TrackerTopology.cc#L50-L67